android/lib/src/main/java/com/ern/api/impl/navigation/ElectrodeCoreActivity.java

```diff
@@ -0,0 +1,164 @@
+package com.ern.api.impl.navigation;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.KeyEvent;
+import android.view.MenuItem;
+import android.view.View;
+
+import androidx.annotation.CallSuper;
+import androidx.annotation.LayoutRes;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.StringRes;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.fragment.app.Fragment;
+
+import com.ern.api.impl.core.ElectrodeReactFragmentActivityDelegate;
+import com.ernnavigationApi.ern.model.NavigationBar;
+import com.facebook.react.ReactRootView;
+
+import org.json.JSONObject;
+
+public abstract class ElectrodeCoreActivity extends AppCompatActivity implements ElectrodeReactFragmentActivityDelegate.DataProvider, MiniAppNavRequestListener {
+    public static final int DEFAULT_TITLE = -1;
+
+    protected ElectrodeReactFragmentActivityDelegate mElectrodeReactNavDelegate;
+
+    /**
+     * Override and provide the main layout resource.
+     *
+     * @return int {@link LayoutRes}
+     */
+    @LayoutRes
+    protected abstract int mainLayout();
+
+    /**
+     * Override to provide title for the landing page.
+     *
+     * @return int @{@link StringRes}
+     */
+    @StringRes
+    protected int title() {
+        return DEFAULT_TITLE;
+    }
+
+
+    @Override
+    @CallSuper
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(mainLayout());
+
+        mElectrodeReactNavDelegate = new ElectrodeReactFragmentActivityDelegate(this);
+        this.getLifecycle().addObserver(mElectrodeReactNavDelegate);
```

Comment:
I know it was copy+pasted, but since the file is new here, you can also just remove `this.` here - not needed.